### ISSUE TITLE
Check on parts component loader

### ISSUE THREAD

**Issue by barke042** [ArctosDB/arctos#5454]
**Observation:**
I am uploading a ton of parts for JFBM:Fish, and it has been chugging along, but the component loader has been stuck for the last few hours.

**Request:**
Possibly other things are taking priority but it would be nice if somebody could kick the tires?

**Comment by dustymc** [comment]
<img width="456" alt="Screenshot 2023-01-06 at 3 10 29 PM" src="https://user-images.githubusercontent.com/5720791/211114868-cee5dc1c-2f7f-4bd5-a68c-fe509ef1bef5.png">

**Comment by barke042** [comment]
Thanks, I have looked there but I'm uncertain how to interpret what I see. Specifically, does the ranking mean that progress ONLY occurs on the highest ranking items and then works down the list? Or can lower priority items be processed simultaneously along with higher priorities? Yesterday when I looked, there were ~2k "Unload Attributes" (a higher priority than parts); if those HAVE to get processed before the parts given the ranking, then that would explain it.

Related, is it very rude for me to put in 40k parts records (currently ranked third priority)? If so, my apologies. What is protocol, should I add the 40k and move the priority down (not sure if I have that ability, I haven't tried)? Or should I break the upload into chunks (which is doable although tedious)?

**Comment by dustymc** [comment]
Reopening for inline documentation help with the text would be very appreciated. Here's what's there now:

``The Component Loader processes all bulkloads other than catalog records. The component loader system is designed to allow processing any number of records from any number of tools without using excessive resources. Some processes may require a significant amount of time to complete. Please plan accordingly.``

``Times below are given in minutes and are approximations. Manage_records users may adjust run order; other updates require an Issue (and possibly data for testing). All loaders are eligible for tuning, those without remarks are probably running at a default safe speed.``

``Drag order (and save) to adjust run order. Be courteous; open an Issue if there are potential conflicts. Last update by jegelewicz at {ts '2023-01-06 11:22:17'} ``

>works down the list

Yes, one at a time.

>rude for me to put in 40 ... records 

Nope, that's why the tool exists.

>break the upload into chunks

Nope, that's why the tool works the way it does. If you can stuff it through your internet, we'll take it.

>is it very rude for me to 

Reprioritizing without discussion (and I think "I did this, any objections?" is an acceptable format) approaches rude, I'm not sure what else might.

**Comment by barke042** [comment]
Something like this would have made it clearer for me:

The Component Loader processes all bulkloads other than catalog records. The component loader system is designed to allow **sequential** processing of any number of records from all upload tools without using excessive resources. Some processes may require a significant amount of time to complete. Please plan accordingly.

Processing times below are given in minutes and are approximations. Be aware that records for each upload tool are processed in the current run order, and that subsequent uploads to a tool with higher run order priority may delay processing of your records. Manage_records users may adjust run order; other updates require an Issue (and possibly data for testing). All loaders are eligible for tuning, those without remarks are probably running at a default safe speed.

**Comment by Jegelewicz** [comment]
> Would it be rude?

Whenever I have some small number of something and I see that someone had 40K of something else processing ahead of me, I am inclined to move mine up as it will finish pretty quickly. I did this while @barke042 parts were processing so that I could clear up the UAM:Art duplicate attributes. That is potentially rude, but I also don't want to file an issue every time I reorder stuff so that 100 attributes can load quickly instead of waiting on 40K parts. Probably what I should have done is move the parts back to the top when the attributes were done?

**Comment by barke042** [comment]
I have no problem with pushing small batches ahead of large batches, I was more wondering whether people putting in large batches should proactively lower their priority, especially during the day when others are trying to get routine work done, as opposed to migrating their whole database. It hadn't really occurred to me that items were completed exactly in order rather than effort being distributed proportional to priority or something similar, so I have unintentionally been hogging processing time during data migration.

**Comment by Jegelewicz** [comment]
> proactively lower their priority,

I don't think you need to do that! If someone is in a hurry, they can move their processing up or request someone to do it for them.

> unintentionally been hogging processing time during data migration.

I don't think you've been hogging anything! Data migration is resource intensive - but everyone currently in Arctos migrated at some time, and everyone should understand what's going on. Nobody has complained and those who really need something done NOW have a way to request that - I think it's all OK, especially with the text changes you guys have suggested to the component loader status instructions.

@barke042, I am SUPER impressed with your data migration progress - rock on!

**Comment by barke042** [comment]
It seems to be back now.

**Comment by dustymc** [comment]
Na, component loaders don't break things. (I think that's actually not true, but it should take a decent network and a lot more than 500 of anything to off Arctos.)

I think the "person with three items and card in hand goes before the full cart" thing is perfectly reasonable, but I'm not sure how to articulate it for documentation. Maybe that's not necessary, everyone knows this as soon as they understand that this is a line?

I don't think un-prioritizing is necessary either, as far as I can tell there's nothing predictable about what someone's in a rush for, whatever sort is 'best' today might be terrible tomorrow.

**Comment by barke042** [comment]
Well, given my new understanding of component loader function, I would certainly take the sequential nature of the list into account when adding items to the queue and when considering re-ranking priorities. And I'll understand why my queue stopped ticking down and not post an issue.
